index.d.ts

```diff
@@ -0,0 +1,454 @@
+     * @example
+     * import FileHound from 'filehound';
+     *
+     * const filehound = FileHound.create();
+     * filehound
+     *  .includeFileStats()
+     *  .find()
+     *  .each(console.log); // array of file objects containing `path` and `stats` properties
+     */
+    public includeFileStats(): FileHound;
+    
+    /**
+     * Find sub-directories
+     *
+     * @memberof FileHound
+     * @instance
+     * @method directory
+     * @returns a FileHound instance
+     * @example
+     * import FileHound from 'filehound';
+     *
+     * const filehound = FileHound.create();
+     * filehound
+     *  .directory()
+     *  .find()
+     *  .each(console.log); // array of matching sub-directories
+     */
+    public directory(): FileHound;
+    
+    /**
+     * Find sockets
+     *
+     * @memberof FileHound
+     * @instance
+     * @method socket
+     * @returns a FileHound instance
+     * @example
+     * import FileHound from 'filehound';
+     *
+     * const filehound = FileHound.create();
+     * filehound
+     *  .socket()
+     *  .find()
+     *  .each(console.log); // array of matching sockets
+     */
+    public socket(): FileHound;
+    
+    /**
+     * Specify the directory search depth. If set to zero, recursive searching
+     * will be disabled
+     *
+     * @memberof FileHound
+     * @instance
+     * @method depth
+     * @returns a FileHound instance
+     * @example
+     * import FileHound from 'filehound';
+     *
+     * const filehound = FileHound.create();
+     * filehound
+     *  .depth(0)
+     *  .find()
+     *  .each(console.log); // array of files names only in the current directory
+     */
+    public depth(depth: number): FileHound;
+    
+    /**
+     * Asynchronously executes a file search.
+     *
+     * @memberof FileHound
+     * @instance
+     * @method find
+     * @param {function} function - Optionally accepts a callback function
+     * @returns Returns a Promise of all matches. If the Promise fulfils,
+     * the fulfilment value is an array of all matching files
+     * @example
+     * import FileHound from 'filehound';
+     *
+     * const filehound = FileHound.create();
+     * filehound
+     *  .find()
+     *  .each(console.log);
+     *
+     * // using a callback
+     * filehound
+     *  .find((err, files) => {
+     *     if (err) return console.error(err);
+     *     console.log(files);
+     *  });
+     */
+    public find(callback?: (err: Error, files: string[]) => void): Promise<string[]>;
+      
+    /**
+    * Synchronously executes a file search.
+    *
+    * @memberof FileHound
+    * @instance
+    * @method findSync
+    * @returns Returns an array of all matching files
+    * @example
+    * import FileHound from 'filehound';
+    *
+    * const filehound = FileHound.create();
+    * const files = filehound.findSync();
+    * console.log(files);
+    */
+    public findSync(): string[];
+    
+    /**
+    * @memberof _ (lodash)
```

Comment:
`each` is a method on the promise returned but `find`. It's not actually a method on `filehound`

Reply:
thanks a lot for the clarification